src/OrleansRuntime/Versions/GrainVersionStore.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.Runtime.Configuration;
+using Orleans.Storage;
+using Orleans.Versions;
+using Orleans.Versions.Compatibility;
+using Orleans.Versions.Selector;
+
+namespace Orleans.Runtime.Versions
+{
+    internal class GrainVersionStore : IVersionStore
+    {
+        private readonly IInternalGrainFactory grainFactory;
+        private readonly string deploymentId;
+        private IVersionStoreGrain StoreGrain => this.grainFactory.GetGrain<IVersionStoreGrain>(this.deploymentId);
```

Comment:
Got it - lgtm. As an aside: shouldn't grain storage be isolated by deployment id anyway? Different deployment = different logical system. It's not something enforced by the system, though, so maybe that's the issue.